dpx/src/dpx_t1_char.rs

```diff
@@ -173,24 +172,18 @@ static mut ps_arg_stack: [f64; 194] = [0.; 194];
  *  Convert ghost hint to edge hint, Counter control for hstem3/vstem3.
  */
 #[inline]
-unsafe extern "C" fn stem_compare(mut v1: *const libc::c_void, mut v2: *const libc::c_void) -> i32 {
-    let mut cmp;
-    let s1 = v1 as *const t1_stem;
-    let s2 = v2 as *const t1_stem;
+fn stem_compare(s1: &t1_stem, s2: &t1_stem) -> ::core::cmp::Ordering {
+    use ::core::cmp::Ordering;
+    // the order of comparing : dir, pos, del
     if (*s1).dir == (*s2).dir {
```

Comment:
Now that you have a Rust &reference, you no longer need to `(*deref).explicitly` — Rust will auto-deref. `s1.dir` will work and be a lot cleaner.

Reply:
(I believe raw pointers don’t have this because dereferencing them is an unsafe operation, so you’re forced to be explicit about it.)